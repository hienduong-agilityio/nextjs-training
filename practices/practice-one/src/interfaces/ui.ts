export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
