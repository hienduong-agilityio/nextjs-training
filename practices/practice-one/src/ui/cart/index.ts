export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
