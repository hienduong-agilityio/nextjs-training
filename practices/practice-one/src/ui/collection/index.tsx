export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
