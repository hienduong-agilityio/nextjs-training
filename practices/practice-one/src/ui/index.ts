export * from './products/index';
export * from './collection/index';
