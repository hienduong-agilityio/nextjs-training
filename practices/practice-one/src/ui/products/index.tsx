export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export * from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProduct';
export * from './ProductDetailTabs';
export * from './ProductReview';
export * from './ProductActions';
export * from './ProductLoading';
