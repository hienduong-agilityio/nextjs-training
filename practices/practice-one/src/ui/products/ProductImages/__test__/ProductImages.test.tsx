import { render, screen } from '@testing-library/react';

// UI
import { ProductImages } from '@/ui';

describe('ProductImages Component', () => {
  const defaultProps = {
    images: ['/images/image-placeholder.svg'],
    name: 'image',
  };

  const renderProductImages = (props = defaultProps) => {
    return render(<ProductImages {...props} />);
  };

  const getMainImage = (altText: string) => screen.getByAltText(altText);

  it('matches the snapshot', () => {
    const { asFragment } = renderProductImages();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with default props', () => {
    renderProductImages();

    expect(screen.queryByAltText(/Thumbnail/)).toBeNull();

    const mainImage = getMainImage(defaultProps.name);
    expect(mainImage).toBeInTheDocument();
    expect(mainImage).toHaveAttribute('src', defaultProps.images[0]);

    const images = ['/images/image1.jpg', '/images/image2.jpg'];
    renderProductImages({ images, name: 'Styled Product' });
  });
});
