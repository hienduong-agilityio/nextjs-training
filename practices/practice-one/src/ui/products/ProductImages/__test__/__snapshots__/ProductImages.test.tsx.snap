// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImages Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid gap-6"
  >
    <div
      class="w-full h-[300px] sm:h-[350px] bg-secondary-50 flex items-center justify-center"
    >
      <img
        alt="image"
        class="object-contain h-[273px] w-[200px] sm:w-[300px] sm:h-[273px] mx-auto"
        data-nimg="1"
        decoding="async"
        height="0"
        loading="lazy"
        src="/images/image-placeholder.svg"
        style="color: transparent;"
        width="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
